packages/dart_frog_cli/lib/src/dev_server_runner/dev_server_runner.dart

```diff
@@ -0,0 +1,249 @@
+
+  Future<void> _codegen() async {
+    logger.detail('[codegen] running pre-gen...');
+    var vars = <String, dynamic>{'port': port};
+    await devServerBundleGenerator.hooks.preGen(
+      vars: vars,
+      workingDirectory: workingDirectory.path,
+      onVarsChanged: (v) => vars = v,
+    );
+
+    logger.detail('[codegen] running generate...');
+    final _ = await devServerBundleGenerator.generate(
+      _target,
+      vars: vars,
+      fileConflictResolution: FileConflictResolution.overwrite,
+    );
+    logger.detail('[codegen] complete.');
+  }
+
+  Future<void> _reload() async {
+    logger.detail('[codegen] reloading...');
+    _isReloading = true;
+    await _codegen();
+    _isReloading = false;
+    logger.detail('[codegen] reload complete.');
+  }
+
+  Future<void> _killProcess(io.Process process) async {
+    _isReloading = false;
+    logger.detail('[process] killing process...');
+    if (_isWindows) {
+      logger.detail('[process] taskkill /F /T /PID ${process.pid}');
+      await _runProcess('taskkill', ['/F', '/T', '/PID', '${process.pid}']);
+    } else {
+      logger.detail('[process] process.kill()...');
+      process.kill();
+    }
+    logger.detail('[process] killing process complete.');
+  }
+
+  // TODO(renancaraujo): this method returns a future that completes when the
+  // process is killed, but it should return a future that completes when the
+  // process is finished starting.
+  /// Starts the development server.
+  Future<ExitCode> start() async {
+    var isHotReloadingEnabled = false;
+
+    Future<void> serve() async {
+      final enableVmServiceFlag = '--enable-vm-service=$dartVmServicePort';
+
+      logger.detail(
+        '''[process] dart $enableVmServiceFlag ${path.join('.dart_frog', 'server.dart')}''',
+      );
+
+      final process = await _startProcess(
+        'dart',
+        [enableVmServiceFlag, path.join('.dart_frog', 'server.dart')],
+        runInShell: true,
+      );
+
+      // On Windows listen for CTRL-C and use taskkill to kill
+      // the spawned process along with any child processes.
+      // https://github.com/dart-lang/sdk/issues/22470
+      if (_isWindows) _sigint.watch().listen((_) => _killProcess(process));
+
+      var hasError = false;
+      process.stderr.listen((_) async {
+        hasError = true;
+
+        if (_isReloading) return;
+
+        final message = utf8.decode(_).trim();
+        if (message.isEmpty) return;
+
+        final isDartVMServiceAlreadyInUseError =
+            _dartVmServiceAlreadyInUseErrorRegex.hasMatch(message);
+        final isSDKWarning = _warningRegex.hasMatch(message);
+
+        if (isDartVMServiceAlreadyInUseError) {
+          logger.err(
+            '$message '
+            '''Try specifying a different port using the `--dart-vm-service-port` argument when running `dart_frog dev`.''',
+          );
+        } else if (isSDKWarning) {
+          /// Do not kill the process if the error is a warning from the SDK.
```

Comment:
```suggestion
          // Do not kill the process if the error is a warning from the SDK.
```

Reply:
lol, in my defense this was like that ion the dev command